src/cmd/csi/provisioner/builder.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/csi/metadata"
+	csiprovisioner "github.com/Dynatrace/dynatrace-operator/src/controllers/csi/provisioner"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const use = "csi-provisioner"
+
+var (
+	probeAddress = ""
+)
+
+type CommandBuilder struct {
+	configProvider  config.Provider
+	managerProvider cmdManager.Provider
+	namespace       string
+	filesystem      afero.Fs
+	csiOptions      *dtcsi.CsiOptions
+}
+
+func NewCsiProvisionerCommandBuilder() CommandBuilder {
+	return CommandBuilder{}
+}
+
+func (builder CommandBuilder) SetConfigProvider(provider config.Provider) CommandBuilder {
+	builder.configProvider = provider
+	return builder
+}
+
+func (builder CommandBuilder) setManagerProvider(provider cmdManager.Provider) CommandBuilder {
+	builder.managerProvider = provider
+	return builder
+}
+
+func (builder CommandBuilder) SetNamespace(namespace string) CommandBuilder {
+	builder.namespace = namespace
+	return builder
+}
+
+func (builder CommandBuilder) setCsiOptions(csiOptions dtcsi.CsiOptions) CommandBuilder {
+	builder.csiOptions = &csiOptions
+	return builder
+}
+
+func (builder CommandBuilder) setFilesystem(filesystem afero.Fs) CommandBuilder {
+	builder.filesystem = filesystem
+	return builder
+}
+
+func (builder CommandBuilder) getCsiOptions() dtcsi.CsiOptions {
+	if builder.csiOptions == nil {
+		builder.csiOptions = &dtcsi.CsiOptions{
+			RootDir: dtcsi.DataPath,
+		}
+	}
+
+	return *builder.csiOptions
+}
+
+func (builder CommandBuilder) getManagerProvider() cmdManager.Provider {
+	if builder.managerProvider == nil {
+		builder.managerProvider = newCsiDriverManagerProvider(probeAddress)
+	}
+
+	return builder.managerProvider
+}
+
+func (builder CommandBuilder) getFilesystem() afero.Fs {
+	if builder.filesystem == nil {
+		builder.filesystem = afero.NewOsFs()
+	}
+
+	return builder.filesystem
+}
+
+func (builder CommandBuilder) Build() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  use,
+		RunE: builder.buildRun(),
+	}
+
+	addFlags(cmd)
+
+	return cmd
+}
+
+func addFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&probeAddress, "health-probe-bind-address", ":10090", "The address the probe endpoint binds to.")
+}
+
+func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		unix.Umask(0000)
+
+		kubeConfig, err := builder.configProvider.GetConfig()
+		if err != nil {
+			return err
+		}
+
+		csiManager, err := builder.getManagerProvider().CreateManager(builder.namespace, kubeConfig)
+		if err != nil {
+			return err
+		}
+
+		err = createCsiDataPath(builder.getFilesystem())
+		if err != nil {
+			return err
+		}
+
+		// TODO: make the code below testable and test it, but in another ticket because otherwise adding the other commands will take a week
```

Comment:
this comment was copied over. wouldn't now be the perfect opportunity to do this and make it testable?

Reply:
I think it should be addressed in separate story. 
Current story is not about refactoring, but feature implementation. Mixing things would lead to problems with e.g. reverting them if needed.

Reply:
this `TODO` is useless because its too generic, "make it testable" tells me nothing, the original writer of the comment must have understood it, but its only confusing at the moment.

also adding a `TODO` comment that says `in another ticket` is a big nono IMO, just create the said ticket, and be done with it.

Reply:
pls remove useless TODO 😬